aten/src/ATen/native/Sorting.cpp

```diff
@@ -182,13 +183,23 @@ void quantile_impl(
   TORCH_CHECK(
       sorted.size(-1) <= std::pow(2, 24),
       "quantile() input tensor is too large");
-  
-  Tensor ranks = q * (sorted.size(-1) - 1);
+
+  Tensor ranks;
+  if (ignore_nan) {
+    ranks = q * (sorted.isnan().logical_not().sum(-1, true) - 1);
+    ranks.masked_fill_(ranks == -1, 0);
```

Comment:
Comment explaining this line, too (probably as part of the previous comment). Looking forward to seeing the all-nan elements test.